code/game/machinery/kitchen/cooking_machines/candy.dm

```diff
@@ -13,6 +13,14 @@
 		"Jelly" = /obj/item/weapon/reagent_containers/food/snacks/variable/jelly
 		)
 
+/obj/machinery/cooker/candy/Initialize()
+	. = ..()
+	component_parts = list()
```

Comment:
adding more of this spaghetticode is not good, maybe at least have a create_parts proc to put them inside

I'm not saying go back and fix every machine that already exists, but don't make this problem worse by doing more of this